src/DataTable/Pagination.tsx

```diff
@@ -0,0 +1,478 @@
+    selectPage,
+    selectNextPage,
+    selectPreviousPage,
+  } = usePagination({
+    defaultPageIndex,
+    onChange,
+    pageSize,
+    totalCount,
+  })
+  const truncatedPageCount = pageCount > 2 ? Math.min(pageCount - 2, MAX_TRUNCATED_STEP_COUNT) : 0
+  const [offsetStartIndex, setOffsetStartIndex] = useState(() => {
+    // Set the offset start index to the page at index 1 since we will have the
+    // first page already visible
+    if (pageIndex === 0) {
+      return 1
+    }
+    return pageIndex
+  })
+  const offsetEndIndex = offsetStartIndex + truncatedPageCount - 1
+  const hasLeadingTruncation = offsetStartIndex >= 2
+  const hasTrailingTruncation = pageCount - 1 - offsetEndIndex > 1
+
+  return (
+    <LiveRegion>
+      <LiveRegionOutlet />
+      <StyledPagination aria-label={label} className="TablePagination" id={id}>
+        <Range pageStart={pageStart} pageEnd={pageEnd} totalCount={totalCount} />
+        <ol className="TablePaginationSteps">
+          <Step>
+            <Button
+              className="TablePaginationAction"
+              type="button"
+              data-has-page={hasPreviousPage ? true : undefined}
+              onClick={() => {
+                if (!hasPreviousPage) {
+                  return
+                }
+
+                selectPreviousPage()
+                if (hasLeadingTruncation) {
+                  if (pageIndex - 1 < offsetStartIndex + 1) {
+                    setOffsetStartIndex(offsetStartIndex - 1)
+                  }
+                }
+              }}
+            >
+              {hasPreviousPage ? <ChevronLeftIcon /> : null}
+              <span className="TablePaginationActionLabel">Previous</span>
+              <VisuallyHidden>&nbsp;page</VisuallyHidden>
+            </Button>
+          </Step>
+          {pageCount > 0 ? (
+            <Step>
+              <Page
+                active={pageIndex === 0}
+                onClick={() => {
+                  selectPage(0)
+                  if (pageCount > 1) {
+                    setOffsetStartIndex(1)
+                  }
+                }}
+              >
+                {1}
+              </Page>
+            </Step>
+          ) : null}
+          {pageCount > 2
+            ? Array.from({length: truncatedPageCount}).map((_, i) => {
+                if (i === 0 && hasLeadingTruncation) {
+                  return <TruncationStep key={`truncation-${i}`} />
+                }
+
+                if (i === truncatedPageCount - 1 && hasTrailingTruncation) {
+                  return <TruncationStep key={`truncation-${i}`} />
+                }
+
+                const page = offsetStartIndex + i
+                return (
+                  <Step key={i}>
+                    <Page
+                      active={pageIndex === page}
+                      onClick={() => {
+                        selectPage(page)
+                      }}
+                    >
+                      {page + 1}
```

Comment:
Something we do in the existing Pagination component is to append a visually hidden "..." after a page number so that screen readers change their tone when reading it.

https://github.com/primer/react/blob/main/src/Pagination/model.tsx#L182-L185

Reply:
Great point, let me make sure that gets added in!

Reply:
Just added, thanks again for bringing this up!